### PR TITLE
KAFKA-13298: Improve documentation on EOS KStream requirements

### PR DESCRIPTION
**Pull request by cddr**
Minor documentation fix to address: https://issues.apache.org/jira/browse/KAFKA-13298

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/streams/developer-guide/config-streams.html [hunk 1]
```diff
@@ -742,7 +742,12 @@ <h4><a class="toc-backref" href="#id30">probing.rebalance.interval.ms</a><a clas
               and <code class="docutils literal"><span class="pre">transaction.state.log.min.isr</span></code>
               to the number of brokers you want to use.
               For more details see <a href="../core-concepts#streams_processing_guarantee">Processing Guarantees</a>.
-            </div></blockquote>
+
+              <dl class="docutils">
+                <dt>Recommendation:</dt>
+                  <dd>While it is technically possible to use EOS with any replication factor, using a replication factor lower than 3 effectively voids EOS. Thus it is strongly recommended to use a replication factor of 3</dd>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: Users should also configure `min.in.sync.replicas=2` for this case. -- We might also want to point out, that the recommendation applies to all topics (ie, __transaction_state, __consumer_offsets, Kafka Streams internal topics, and user topics).

nit: missing `.` at the end.

**Review by mjsax — COMMENTED** [review]
Thanks for the PR! Some recommendations to be more explicit/precise.

File: docs/streams/developer-guide/config-streams.html (hunk 1 above)
Comment (cddr) [reply docs/streams/developer-guide/config-streams.html]: Thanks for the review @mjsax. I have updated the PR accordingly.

File: docs/streams/developer-guide/config-streams.html [hunk 2]
```diff
@@ -742,7 +742,12 @@ <h4><a class="toc-backref" href="#id30">probing.rebalance.interval.ms</a><a clas
               and <code class="docutils literal"><span class="pre">transaction.state.log.min.isr</span></code>
               to the number of brokers you want to use.
               For more details see <a href="../core-concepts#streams_processing_guarantee">Processing Guarantees</a>.
-            </div></blockquote>
+
+              <dl class="docutils">
+                <dt>Recommendation:</dt>
+                  <dd>While it is technically possible to use EOS with any replication factor, using a replication factor lower than 3 effectively voids EOS. Thus it is strongly recommended to use a replication factor of 3 (together with min.in.sync.replicas=2). This recommendation applies to all topics (i.e. __transaction_state, __consumer_offsets, Kafka Streams internal topics, and user topics).</dd>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: Can we break this line into multiple? Hard to review long lines...

Should we use some markup for `__consumer_offsets` (etc) ?

**Comment by mjsax** [comment]
Thanks for the PR @cddr! Merged to `trunk`.

**Pull request merged**